core/src/main/java/io/spine/core/Commands.java

```diff
@@ -106,12 +107,88 @@ public static Message ensureMessage(Message commandOrMessage) {
      */
     public static TenantId getTenantId(Command command) {
         checkNotNull(command);
-        final TenantId result = command.getContext()
-                                       .getActorContext()
-                                       .getTenantId();
+        final TenantId result = getTenantId(command.getContext());
         return result;
     }
 
+    /**
+     * Obtains a {@link TenantId} from the given {@link Event}.
+     *
+     * <p>The {@code TenantId} is retrieved by traversing the passed {@code Event}s context. It is
+     * stored in the initial {@link CommandContext} and can be retrieved from the events origin 
+     * command or rejection context. 
+     */
+    @Internal
+    public static TenantId getTenantId(Event event) {
```

Comment:
What I was commenting about “guts” of `TenantId` and the `CommandContext` was related to the method with `CommandContext` parameter. 

This method, which accepts `Event`, looks like a part of `Events` utility class.